torch/fx/proxy.py

```diff
@@ -259,6 +270,26 @@ def node(self):
     def __call__(self, *args, **kwargs):
         return self.tracer.create_proxy('call_method', self.attr, (self.root,) + args, kwargs)
 
+
+class ParameterProxy(Proxy):
+    """
+    a special proxy which lets "shape" attribute accesses pass through to the underlying
+    module parameter object, so that conditional tests on parameter shapes can be traced
+    """
+    def __init__(self, tracer: TracerBase, node: Node, name, param):
+        super().__init__(node, tracer)
+        assert(isinstance(param, torch.nn.Parameter))
+        self.param = param
+        self.name = name
+
+    def __repr__(self) -> str:
+        return f'ParameterProxy({self.name})'
+
+    @property
+    def shape(self):
+        return self.param.shape
+
```

Comment:
A few more methods/properties we could forward:
* [dim](https://pytorch.org/docs/stable/generated/torch.Tensor.dim.html#torch.Tensor.dim)
* [ndim](https://pytorch.org/docs/stable/generated/torch.Tensor.ndim.html#torch.Tensor.ndim)
* [size](https://pytorch.org/docs/stable/generated/torch.Tensor.size.html#torch.Tensor.size)
* [numel](https://pytorch.org/docs/stable/generated/torch.Tensor.numel.html#torch.Tensor.numel)
* [nelement](https://pytorch.org/docs/stable/generated/torch.Tensor.nelement.html#torch.Tensor.nelement)

I think it's also fair to consider whether dtype/device/strides should be usable in this context, but we can just start off with the shape-related properties in this PR

Reply:
added these properties